src/System.Private.CoreLib/shared/System/Diagnostics/Tracing/CounterGroup.cs

```diff
@@ -166,12 +168,30 @@ private void EnableTimer(float pollingIntervalInSeconds)
                 }
             }
             // Always fire the timer event (so you see everything up to this time).  
-            OnTimer(null);
+            OnTimer();
         }
 
-        private void OnTimer(object? state)
+        private void PollForValues()
         {
-            Debug.WriteLine("Timer fired at " + DateTime.UtcNow.ToString("mm.ss.ffffff"));
+            while(true)
+            {
+                while(_isPollingEnabled)
+                {
+                    if (!OnTimer())
+                    {
+                        break;
+                    }
+                    Thread.Sleep(_pollingIntervalInMilliseconds);
+                }
+
+                _pollingEnabledEvent.WaitOne(); // Block until polling is enabled again
+                _pollingEnabledEvent.Reset(); // Reset this.
+            }
+        }
+
+        private bool OnTimer()
+        {
+            Debug.WriteLine("Polling thread fired at " + DateTime.UtcNow.ToString("mm.ss.ffffff"));
```

Comment:
We do not use unconditional debug outputs like this. It gets pretty annoying when one debugs unrelated problem. If you really want to have these around, you should keep them commented out or under define, and only enable this define for your local builds.